block-header-gossip.md

```diff
@@ -0,0 +1,198 @@
+# Block header gossip 
+The document describes how the block information (accumulator & block headers) are exchanged between the portal network clients. 
+
+Each client has the responsibilty to track the tip of the chain and to store accumulators and block header information locally. Clients will be sharing the information with requesting clients in the network to maintain the health of the network as a whole
+
+
+## Block storage
+Each client will be storing blocks in two forms. As accumulators where the entire block history is recorded and as partial block headers where only a subset of the block header is stored. Partial block headers are only stored for the most recent N number of blocks.
+
+### Accumulator
+Portal network's accumulator will be based on the [double-batched merkle log accumulator](https://ethresear.ch/t/double-batched-merkle-log-accumulator/571) 
+
+#### Epoch Accumulator
+A fixed sized accumulator of a proposed length of 2048. Block info will be added in into the SSZ list until it has filled up all 2048 of its entries. After which the next block will be included in a new epoch accumulator.
+
+SSZ sede structure:
+`List[Container[blockhash:bytes32, total_difficulty:uint256], max_length=2048]`
+
+The root hash of an epoch will then be included as an entry in the master accumulator.
+
+Aside from the first epoch, each client will need to store the current epoch as well as the epoch before in case of reorgs and block sync.
+
+#### Master Accumulator
+
+An accumulator that increases in size with each epoch. The root hash of each epoch will be appended to the master accumulator
+
+SSZ sede structure:
+`List[epoch_root_hash:bytes32, max_length=<a_large_number>]`
+
+
+### Partial Block Header
+
+Clients will need more information to verify the validity of the block. In the event of network latency, some clients may have missed out on the last few blocks and will need to request for the partial block header from its neighbours. 
+
+And so, each client number will also be required to store a certain amount of the most recent blocks.
+
+Partial Block Header will include:
+- Block Number
+- Block Hash
+- Difficulty
+- Nonce
```

Comment:
I think that we should have a justification for each of these fields.  I'm curious to know the reasoning behind inclusion of number, difficulty, nonce, mixhash, and previous hash.  I would also advocate for adding "total difficulty" which is not a header field but also something that can't be computed without processing the full chain.

- block number can be derived from the actual accumulator information (number of entries in the master, and position in the epoch)
- nonce and mixhash aren't obvious to me why they are needed.
- previous hash can generally be found in the accumulator info with the exception of at epoch boundaries, and in general it can be retrieved from the actual header itself.

Reply:
oh what I wanted to achieve for partial block header will be for each client to have the ability to verify the POW seal and to append valid blocks to their local accumulator store. Each client will receive latest blocks from its neighbours. and so I believe the blocknumber, difficulty, nonce, mixhash and previous hash will be required. My idea for previous hash also includes a strategy to handle for reorgs

My idea is for the partial block header to be passed around only at the latest blocks when the client needs to check if the block is valid. If the block is valid and is in the canonical chain, the client can include total_difficulty in the accumulator